integration_tests/tests/admin/userManagement.cy.ts

```diff
@@ -335,4 +335,27 @@ context('User management', () => {
     // Then the page should show the results for the second page
     page.shouldShowUsers(usersForResultsPage2)
   })
+
+  it('retrieve the updated user from api if version not matched to user in session and shows the new role added report viewer', () => {
```

Comment:
This test might be clearer if 2 successive API calls are stubbed with a different `x-cas-user-version` header, to show that there's a change. At the moment there is so much interaction around users it muddies the readability -- I would suggest 2 successive visits of the same page (even the home page would be enough) should be sufficient.